### PR TITLE
Bring markdown support to collection and dashboard descriptions

### LINKED ISSUE #28754: [Epic] Bring Markdown to descriptions of collections, questions, dashboards
We currently support autolinking link-like strings (e.g. [google.com](http://google.com/), https://google.com/, [a@google.com](mailto:a@google.com)) in Markdown on dashboard text cards, but not in descriptions (collections, questions, dashboards), which are just plain text. We should add markdown support for these entities.

**Implementation Plan**
We already have a `Markdown` component used to render descriptions of timelines and timeline events: https://github.com/metabase/metabase/blob/6426f366da599dca80f0165da1e7dc040f9d2d4c/frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.tsx#L39 https://github.com/metabase/metabase/blob/24c7832edfa440793d0ddc5b36f29ecbdd9bf84e/frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.styled.tsx#L29 We could reuse it to render the description of other entities. And we could continue editing the description with a plain `TextArea` component as we do for timelines.

```[tasklist]
- [ ] https://github.com/metabase/metabase/pull/29753
- [ ] https://github.com/metabase/metabase/pull/29624
- [ ] https://github.com/metabase/metabase/pull/29624
```

**How to Test This Feature Out**

- Add a link with the markdown syntax to collection/question/dashboard description.
- Make sure it is rendered as a link.

### PR DESCRIPTION
**Pull request by uladzimirdev**
It's first part of the #28754 regarding collection

### Description

Bring markdown support to collection description

### How to verify

1. New Collection -> put markdown like `**important** text` to the description -> go the the collection root and check tooltip

### Demo

Screenshot 2023-03-27 at 11 14 48

### Checklist

- [x] Tests have been added/updated to cover changes in this PR
No unit tests found, but added e2e

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29624)

### CODE REVIEW

File: e2e/test/scenarios/collections/collections.cy.spec.js
Comment (kulyk) [inline e2e/test/scenarios/collections/collections.cy.spec.js:1]: What do we gain here by covering it with Cypress? I think we could achieve the same thing with the testing-library?

File: e2e/test/scenarios/collections/collections.cy.spec.js [hunk 1]
```diff
@@ -178,12 +178,48 @@ describe("scenarios > collection defaults", () => {
...
+    it('should support markdown in collection description', () => {
+      getCollectionIdFromSlug("first_collection", id => {
+        visitCollection(id);
+      });
+
+      cy
+        .findByPlaceholderText("Add description")
+        .clear()
+        .type("**important text** and [link](https://metabase.com)")
+        .blur();
```
Comment (kulyk) [inline e2e/test/scenarios/collections/collections.cy.spec.js:205]: How important is it that we're adding a description manually? We could cut some CI time if we create an item we need via API request

File: e2e/test/scenarios/collections/collections.cy.spec.js [hunk 2]
```diff
@@ -178,12 +178,48 @@ describe("scenarios > collection defaults", () => {
...
+      cy.get('table').within(() => {
+        cy.findByText("First collection").parent().parent().within(() => {
+          cy.icon("info").trigger("mouseenter"); 
+        })
+      })
+    
+      popover().within(() => {
+        cy.findByText(/important text/);
+        cy.findByText(/link/);
+      })
```
Comment (kulyk) [inline e2e/test/scenarios/collections/collections.cy.spec.js:218]: These assertions would pass if we revert the `<Markdown />` component inside the `BaseTableItem`. I'm not sure there's a good way to guarantee formatting is correct, but maybe we could also assert there are no asterisks or brackets? Hopefully that'd give us a bit more confidence here

File: e2e/test/scenarios/collections/collections.cy.spec.js
Comment (kulyk) [reply e2e/test/scenarios/collections/collections.cy.spec.js:1]: @uladzimirdev fwiw we actually do have [a jest test suite for it](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/collections/components/BaseItemsTable.jsx)

File: e2e/test/scenarios/collections/collections.cy.spec.js (hunk 1 above)
Comment (kulyk) [reply e2e/test/scenarios/collections/collections.cy.spec.js:205]: There is, we try to keep an eye on that and use the API whenever it makes sense. It's pretty fast locally, but when we're running the full test suite in the CI environment, optimizations like that can make a difference when stacked together

File: e2e/test/scenarios/collections/collections.cy.spec.js
Comment (uladzimirdev) [reply e2e/test/scenarios/collections/collections.cy.spec.js:1]: oh, really, I missed it, thanks

**Review by WiNloSt — CHANGES_REQUESTED** [review]
Note: I reviewed with reviewable.io. Not sure how thing's are going to pan out though.

I add some comments. Please take a look, otherwise, I tested this and it looks good imo.

> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29624#-:-NRhg1ea2rhKDpZwdfGU:bhz888v)* status: all files reviewed, 2 unresolved discussions (waiting on @kulyk)

___
*[`e2e/test/scenarios/collections/collections.cy.spec.js` line 205 at r2](https://reviewable.io/reviews/metabase/metabase/29624#-NRe2hUnAIMlV0E4WU6U-r2-205:-NRhdZ7Y6_GE9g8hINiq:b-yke7wq) ([raw file](https://github.com/metabase/metabase/blob/e90850099134eaa69890a7a4885d0a468df2082c/e2e/test/scenarios/collections/collections.cy.spec.js#L205)):*
<details><summary><i>Previously, kulyk (Anton Kulyk) wrote…</i></summary><blockquote>

There is, we try to keep an eye on that and use the API whenever it makes sense. It's pretty fast locally, but when we're running the full test suite in the CI environment, optimizations like that can make a difference when stacked together
</blockquote></details>

I agree with @kulyk on this. It's not important that we need to test typing out the description, what's more important is how we're displaying it in the tooltip.

I think we should favor using API calls to edit/create description in this case
___
*[`e2e/test/scenarios/collections/collections.cy.spec.js` line 218 at r2](https://reviewable.io/reviews/metabase/metabase/29624#-NRe2hUnAIMlV0E4WU6U-r2-218:-NRhee0X70PNEaCpH2WI:b-yd68eb) ([raw file](https://github.com/metabase/metabase/blob/e90850099134eaa69890a7a4885d0a468df2082c/e2e/test/scenarios/collections/collections.cy.spec.js#L218)):*
<details><summary><i>Previously, uladzimirdev (Uladzimir Havenchyk) wrote…</i></summary><blockquote>

good point, it should be possible to assert exactly link, I'll fix it
</blockquote></details>

As per our discussion on [Slack here](https://metaboat.slack.com/archives/C04DAKFDTCZ/p1680096760983479?thread_ts=1680096448.209279&cid=C04DAKFDTCZ), I don't think we should use links in the markdown as an example since there's no way to click that. I mean I know it's technically to render links, but that's not useful since there's no way to click it.

All this on the assumption that the only way to view the description is in the tooltip, please correct me if there's other way we can see the description.

File: e2e/test/scenarios/collections/collections.cy.spec.js
Comment (uladzimirdev) [reply e2e/test/scenarios/collections/collections.cy.spec.js:1]: Added unit test together with e2e

File: e2e/test/scenarios/collections/collections.cy.spec.js (hunk 1 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/collections/collections.cy.spec.js]: replaced DOM interactions with `cy.request`

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js [hunk 3]
```diff
@@ -117,6 +117,43 @@ describe("Collections BaseItemsTable", () => {
...
+  describe("description", () => {
+    it("shows description on hover", () => {
+      const DESCRIPTION = "My collection";
+      const ITEM = getCollectionItem({ description: DESCRIPTION });
+
+      setup({ items: [ITEM] });
+
+      const icon = screen.getByRole("img", {
+        name: /info icon/i,
+      });
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js:129]: You can use a `getIcon` helper from `__support__/ui`:

```tsx
const icon = getIcon("info");
screen.hover(icon);
```

**Review by WiNloSt — APPROVED** [review]
This looks good to me now :thumbsup: 

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29624#-:-NRmKmyR1A6Cs2p-UCAB:b-vismlr)* status: 1 of 3 files reviewed, 2 unresolved discussions (waiting on @kulyk)

File: frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js [hunk 4]
```diff
@@ -117,6 +117,43 @@ describe("Collections BaseItemsTable", () => {
...
+      userEvent.hover(icon);
+
+      const tooltip = screen.getByRole("tooltip");
+
+      expect(tooltip).toBeInTheDocument();
+      expect(tooltip).toHaveTextContent("important text");
+      expect(tooltip).not.toHaveTextContent(DESCRIPTION);
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js:153]: I think `toHaveTextContent("important text")` will fail if it's `**important** text`? I.e. seems like we don't need the `not.toHaveTextContext` assertion

File: e2e/test/scenarios/collections/collections.cy.spec.js [hunk 5]
```diff
@@ -178,12 +178,26 @@ describe("scenarios > collection defaults", () => {
...
+    cy.request("PUT", "/api/collection/9", {
+      description: "# header"
+    })
+
+    visitRootCollection();
+
+    cy.get('table').within(() => {
+      cy.findByText("First collection").parent().parent().within(() => {
+        cy.icon("info").trigger("mouseenter"); 
+      })
```
Comment (kulyk) [inline e2e/test/scenarios/collections/collections.cy.spec.js:191]: Do we really need to grab `parent().parent()` here? Are there more than one "info" icons in scope of `findByText("First collection")`

**Review by kulyk — COMMENTED** [review]
Looks good! Left a few questions & tips about testing helpers. @uladzimirdev could you please rebase the branch onto the latest `master` and run `yarn prettier`? It looks like e2e code isn't formatted properly

File: e2e/test/scenarios/collections/collections.cy.spec.js (hunk 5 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/collections/collections.cy.spec.js:191]: yup, almost every line contains an icon, I need first, but if we change the structure of tests, it will fail if we don't use specific line. I'll take a look if it can be simplified

**Comment by uladzimirdev** [comment]
> could you please rebase the branch onto the latest `master` and run `yarn prettier`? It looks like e2e code isn't formatted properly

actually I did it, let me try again

File: frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js:153]: you're right, I overcomplicated it

Comment (uladzimirdev) [reply frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js]: fixed

File: e2e/test/scenarios/collections/collections.cy.spec.js (hunk 5 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/collections/collections.cy.spec.js]: used `closest('tr')` instead

**Review by kulyk — APPROVED** [review]

File: frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js]: screen.hover is not a function. I presume you meant `userEvent.hover`

Comment (kulyk) [reply frontend/src/metabase/collections/components/BaseItemsTable.unit.spec.js]: Oh yeah, my bad! `userEvent` for sure
